arroyo/processing/processor.py

```diff
@@ -300,6 +260,47 @@ def on_partitions_revoked(partitions: Sequence[Partition]) -> None:
             [topic], on_assign=on_partitions_assigned, on_revoke=on_partitions_revoked
         )
 
+    def _close_processing_strategy(self) -> None:
+        """Close the processing strategy and wait for it to exit."""
+        start_close = time.time()
+
+        if self.__processing_strategy is None:
+            # Partitions are revoked when the consumer is shutting down, at
+            # which point we already have closed the consumer.
+            return
+
+        logger.info("Closing %r...", self.__processing_strategy)
+        self.__processing_strategy.close()
+
+        logger.info("Waiting for %r to exit...", self.__processing_strategy)
+
+        while True:
+            start_join = time.time()
+
+            try:
+                self.__processing_strategy.join(self.__join_timeout)
+                self.__metrics_buffer.incr_timing(
+                    "arroyo.consumer.join.time", time.time() - start_join
+                )
+                break
```

Comment:
This code may raise MessageRejected. If that happens we would skip these lines 
```
self.__consumer.close()
 self.__processor_factory.shutdown()
```
at line 530

Are you sure that is not a problem ?